<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>BMEditMainWindow</class>
 <widget class="QMainWindow" name="BMEditMainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1462</width>
    <height>915</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string notr="true">BMEdit</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QOpenGLWidget" name="sceneGLView"/>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1462</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <widget class="QMenu" name="menuExport">
     <property name="title">
      <string>Export</string>
     </property>
     <addaction name="actionExport_PRP_properties"/>
    </widget>
    <addaction name="actionOpen_level"/>
    <addaction name="menuExport"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <addaction name="actionTools"/>
    <addaction name="actionScene"/>
    <addaction name="actionProperties"/>
    <addaction name="separator"/>
    <addaction name="actionRestore_layout"/>
    <addaction name="separator"/>
    <addaction name="actionTypes_Viewer"/>
   </widget>
   <widget class="QMenu" name="menuScene">
    <property name="title">
     <string>Scene</string>
    </property>
    <addaction name="actionSave_properties"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuView"/>
   <addaction name="menuScene"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QDockWidget" name="toolsDock">
   <property name="minimumSize">
    <size>
     <width>120</width>
     <height>35</height>
    </size>
   </property>
   <property name="features">
    <set>QDockWidget::DockWidgetClosable|QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="allowedAreas">
    <set>Qt::LeftDockWidgetArea|Qt::RightDockWidgetArea</set>
   </property>
   <property name="windowTitle">
    <string>Tools</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="toolsDockContents"/>
  </widget>
  <widget class="QDockWidget" name="sceneDock">
   <property name="minimumSize">
    <size>
     <width>260</width>
     <height>189</height>
    </size>
   </property>
   <property name="windowTitle">
    <string>Scene</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="sceneDockContents">
    <layout class="QHBoxLayout" name="horizontalLayout_4">
     <item>
      <widget class="QTabWidget" name="sceneTabs">
       <property name="tabShape">
        <enum>QTabWidget::Rounded</enum>
       </property>
       <property name="currentIndex">
        <number>0</number>
       </property>
       <property name="elideMode">
        <enum>Qt::ElideNone</enum>
       </property>
       <property name="documentMode">
        <bool>false</bool>
       </property>
       <property name="tabsClosable">
        <bool>false</bool>
       </property>
       <widget class="QWidget" name="sceneTreeTab">
        <attribute name="title">
         <string>Scene Tree</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <item>
            <widget class="QLabel" name="searchObjectLabel">
             <property name="text">
              <string>Node name:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="searchInputField">
             <property name="enabled">
              <bool>false</bool>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="widgets::SceneTreeView" name="sceneTreeView">
           <property name="editTriggers">
            <set>QAbstractItemView::AllEditTriggers</set>
           </property>
           <property name="rootIsDecorated">
            <bool>true</bool>
           </property>
           <property name="animated">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="scenePropertiesTab">
        <attribute name="title">
         <string>Scene Properties</string>
        </attribute>
        <layout class="QHBoxLayout" name="horizontalLayout_5">
         <item>
          <widget class="QTableView" name="sceneProperties">
           <property name="editTriggers">
            <set>QAbstractItemView::AnyKeyPressed|QAbstractItemView::DoubleClicked|QAbstractItemView::EditKeyPressed|QAbstractItemView::SelectedClicked</set>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="propertiesDock">
   <property name="minimumSize">
    <size>
     <width>260</width>
     <height>327</height>
    </size>
   </property>
   <property name="floating">
    <bool>false</bool>
   </property>
   <property name="features">
    <set>QDockWidget::DockWidgetClosable|QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="windowTitle">
    <string>Properties</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="propertiesDockContents">
    <layout class="QVBoxLayout" name="verticalLayout_4">
     <item>
      <widget class="QTabWidget" name="geomData">
       <property name="currentIndex">
        <number>0</number>
       </property>
       <widget class="QWidget" name="properties">
        <attribute name="title">
         <string>Properties</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout_6">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QLabel" name="label">
             <property name="text">
              <string>Type:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="sceneObjectTypeCombo">
             <property name="enabled">
              <bool>false</bool>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <item>
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>Name:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="sceneObjectName">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="accessibleDescription">
              <string notr="true"/>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>Properties:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QTableView" name="propertiesView">
           <property name="selectionMode">
            <enum>QAbstractItemView::SingleSelection</enum>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="controllers">
        <attribute name="title">
         <string>Controllers</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout_5">
         <item>
          <widget class="widgets::GeomControllersWidget" name="geomControllers" native="true"/>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="primitivesDock">
   <property name="features">
    <set>QDockWidget::DockWidgetFeatureMask</set>
   </property>
   <property name="windowTitle">
    <string>Primitives</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="primitivesDockContents">
    <layout class="QHBoxLayout" name="horizontalLayout_8">
     <item>
      <widget class="QSplitter" name="splitter">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <widget class="QWidget" name="">
        <layout class="QVBoxLayout" name="verticalLayout_7">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_6">
           <item>
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>Primitives count: </string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="primitivesCountLabel">
             <property name="text">
              <string>0</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QPushButton" name="exportChunk">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>Export chunk</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="">
        <layout class="QVBoxLayout" name="verticalLayout_8">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_7">
           <item>
            <widget class="QLabel" name="label_5">
             <property name="text">
              <string>Filter by:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="primitivesFilter_UnknownPrimType">
             <property name="text">
              <string>Unknown</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="primitivesFilter_ZeroBufferPrimType">
             <property name="text">
              <string>Zero buffer</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="primitivesFilter_DescriptionPrimType">
             <property name="text">
              <string>Description Buffer</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="primitivesFilter_IndexBufferPrimType">
             <property name="text">
              <string>Index Buffer</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="primitivesFilter_VertexBufferPrimType">
             <property name="text">
              <string>Vertex Buffer</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QTableView" name="scenePrimitivesTable"/>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="primitivePreviewDock_2">
   <property name="windowTitle">
    <string>Primitive preview</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_2">
    <layout class="QHBoxLayout" name="horizontalLayout_11">
     <item>
      <widget class="widgets::PrimitivePreviewWidget" name="scenePrimitivePreview">
       <property name="minimumSize">
        <size>
         <width>100</width>
         <height>100</height>
        </size>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="actionOpen_level">
   <property name="text">
    <string>Open level</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionTools">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Tools</string>
   </property>
  </action>
  <action name="actionScene">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Scene</string>
   </property>
  </action>
  <action name="actionProperties">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Properties</string>
   </property>
  </action>
  <action name="actionRestore_layout">
   <property name="text">
    <string>Restore layout</string>
   </property>
  </action>
  <action name="actionTypes_list">
   <property name="text">
    <string>Types list</string>
   </property>
  </action>
  <action name="actionTypes_Viewer">
   <property name="text">
    <string>Types Viewer</string>
   </property>
  </action>
  <action name="actionSave_properties">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Save properties</string>
   </property>
  </action>
  <action name="actionExport_PRP_properties">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Export PRP (properties)</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>widgets::GeomControllersWidget</class>
   <extends>QWidget</extends>
   <header>Widgets/GeomControllersWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>widgets::SceneTreeView</class>
   <extends>QTreeView</extends>
   <header>Widgets/SceneTreeView.h</header>
  </customwidget>
  <customwidget>
   <class>widgets::PrimitivePreviewWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>Widgets/PrimitivePreviewWidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
